-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-41118/34755

ERROR: Build errors found during clang-tidy run.
```
PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc:42:40: error: no member named 'GetInputNameAllocated' in 'Ort::Session' [clang-diagnostic-error]
      std::string input_name(session_->GetInputNameAllocated(i, allocator));
                                       ^
PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc:60:41: error: no member named 'GetOutputNameAllocated' in 'Ort::Session' [clang-diagnostic-error]
      std::string output_name(session_->GetOutputNameAllocated(i, allocator));
                                        ^
Suppressed 356 warnings (355 in non-user code, 1 with check filters).
--
gmake: *** [config/SCRAM/GMake/Makefile.coderules:129: code-checks] Error 2
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 2
```

--- a/DQM/EcalMonitorClient/src/MLClient.cc
+++ b/DQM/EcalMonitorClient/src/MLClient.cc
@@ -164,14 +164,14 @@ namespace ecaldqm {
 
     Ort::AllocatorWithDefaultOptions allocator;
 
-    const char* inputName = session.GetInputName(0, allocator);
+    const char* inputName = session.GetInputNameAllocated(0, allocator).get();
 
     Ort::TypeInfo inputTypeInfo = session.GetInputTypeInfo(0);
     auto inputTensorInfo = inputTypeInfo.GetTensorTypeAndShapeInfo();
 
     std::vector<int64_t> inputDims = inputTensorInfo.GetShape();
 
-    const char* outputName = session.GetOutputName(0, allocator);
+    const char* outputName = session.GetOutputNameAllocated(0, allocator).get();
 
     Ort::TypeInfo outputTypeInfo = session.GetOutputTypeInfo(0);
     auto outputTensorInfo = outputTypeInfo.GetTensorTypeAndShapeInfo();

--- a/PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc
+++ b/PhysicsTools/ONNXRuntime/src/ONNXRuntime.cc
@@ -39,7 +39,7 @@ namespace cms::Ort {
 
     for (size_t i = 0; i < num_input_nodes; i++) {
       // get input node names
-      std::string input_name(session_->GetInputNameAllocated(i, allocator));
+      std::string input_name(session_->GetInputNameAllocated(i, allocator).get());
       input_node_strings_[i] = input_name;
       input_node_names_[i] = input_node_strings_[i].c_str();
 
@@ -57,7 +57,7 @@ namespace cms::Ort {
 
     for (size_t i = 0; i < num_output_nodes; i++) {
       // get output node names
-      std::string output_name(session_->GetOutputNameAllocated(i, allocator));
+      std::string output_name(session_->GetOutputNameAllocated(i, allocator).get());
       output_node_strings_[i] = output_name;
       output_node_names_[i] = output_node_strings_[i].c_str();
 

